src/test/feature/lib/hdfs_config.h

```diff
@@ -0,0 +1,172 @@
+     * @return true if getStandbyNamenode succeeded
+     */
+    bool getStandbyNamenode(std::string &standbynamenode,
+                            int &port);
+
+    /**
+     * get HDFS namenode(s) information
+     * @param namenodes, namenodes' hostnames reference which will be set
+     * @param port, namenodes' ports reference which will be set
+     */
+    void getNamenodes(std::vector<std::string> &namenodes,
+                      std::vector<int> &port);
+
+    /**
+     * get HDFS datanodes information
+     * @param datanodelist, datanodes' hostnames reference which will be set
+     * @param port, datanodes' ports reference which will be set
+     */
+    void getDatanodelist(std::vector<std::string> &datanodelist,
+                         std::vector<int> &port);
+
+    /**
+     * get HDFS active datanodes information
+     * @param activedatanodes, active datanodes' hostnames reference which will be set
+     * @param port, active datanodes' ports reference which will be set
+     */
+    void getActiveDatanodes(std::vector<std::string> &activedatanodes,
+                            std::vector<int> &port);
+
+    /**
+     * whether HDFS is in safe mode
+     * @return true if HDFS is in safe node
+     */
+    bool isSafemode();
+
+    /**
+     * get parameter value in ./etc/hdfs-client.xml or ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, used to get parameter value
+     * @param conftype, get parameter value, 'hdfs' or 'HDFS' from ./etc/hdfs-client.xml, others from ./etc/hadoop/hdfs-site.xml
+     * @return parameter value
+     */
+    std::string getParameterValue(const std::string &parameterName, const std::string &conftype);
+    
+    /**
+     * get parameter value in ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, used to get parameter value
+     * @return parameter value
+     */
+    std::string getParameterValue(const std::string &parameterName);
+
+    /**
+     * set parameter value in ./etc/hdfs-client.xml or ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, parameter name which used to set parameter value
+     * @param parameterValue, parameter value which to be set
+     * @param conftype, get parameter value, 'hdfs' or 'HDFS' from ./etc/hdfs-client.xml, others from ./etc/hadoop/hdfs-site.xml
+     * @return true if succeeded
+     */
+    bool setParameterValue(const std::string &parameterName, const std::string &parameterValue, const std::string &conftype);
+
+    /**
+     * set parameter value in ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, parameter name which used to set parameter value
+     * @param parameterValue, parameter value which to be set
+     * @return true if succeeded
+     */
+    bool setParameterValue(const std::string &parameterName, const std::string &parameterValue);
+
+  private:
+    /**
+     * @return yarn user
```

Comment:
TYPO: `return hdfs user`. Actually, I do not suggest that we should add doxygen-style comments of private interfaces.